lib/crossing.rb

```diff
@@ -9,26 +9,27 @@ def initialize(s3_client)
   end
 
   def put(bucket, filename)
-    begin
-      file = File.new(filename, 'r')
-    rescue
-      raise CrossingFileNotFoundException, "File not found: #{filename}"
+    File.open(filename, 'r') do |file|
+      put_content(bucket, File.basename(filename), file.read)
     end
-
-    put_content(bucket, filename.split('/').last, file.read)
+  rescue Errno::ENOENT
```

Comment:
If you can put all of the code for a method in a single begin block, then you can omit begin and simply rescue the entire method. Logically the same, more idiomatic Ruby.

You should attempt to rescue the most specific error that makes sense. In this case, Errno:ENOENT is file not found as thrown by File.open